botocore/docs/example.py

```diff
@@ -48,6 +50,8 @@ def document_recursive_shape(self, section, shape, **kwargs):
     def document_shape_default(self, section, shape, history, include=None,
                                exclude=None, **kwargs):
         py_type = py_default(shape.type_name)
+        if self._context.get('streaming_shape') == shape:
```

Comment:
Is the caching of `streaming_shape` inside of a `self._context` value necessary? Do we need to figure out if it's streaming or not multiple times for a single shape?